app/src/main/java/com/hedvig/app/feature/chat/ui/ChatActivity.kt

```diff
@@ -82,6 +84,44 @@ class ChatActivity : BaseActivity(R.layout.activity_chat) {
             resources.getDimensionPixelSize(R.dimen.is_keyboard_brake_point_height)
         navHeightDiff = resources.getDimensionPixelSize(R.dimen.nav_height_div)
 
+        chatRoot.setEdgeToEdgeSystemUiFlags(true)
+
+        input.doOnApplyWindowInsets { view, insets, initialState ->
+            view.updatePadding(
+                bottom = initialState.paddings.bottom + insets.systemWindowInsetBottom
+            )
+        }
+
+        input.textInputContainer.show()
+
+        input.measure(
+            ConstraintLayout.LayoutParams.MATCH_PARENT,
+            ConstraintLayout.LayoutParams.WRAP_CONTENT
+        )
+
+        frameLayout.measure(
+            ConstraintLayout.LayoutParams.MATCH_PARENT,
+            ConstraintLayout.LayoutParams.WRAP_CONTENT
+        )
+
+        frameLayout.doOnApplyWindowInsets { view, insets, initialState ->
+            view.updatePadding(
+                top = initialState.paddings.top + insets.systemWindowInsetTop
+            )
+        }
+
+        messages.doOnApplyWindowInsets { view, insets, initialState ->
+            view.updatePadding(
+                top = initialState.paddings.top + frameLayout.measuredHeight,
+                bottom = initialState.paddings.bottom + insets.systemWindowInsetBottom + input.textInputContainer.measuredHeight
+            )
+        }
+
+        input.textInputContainer.remove()
```

Comment:
Let's make this procedure a function on the `ChatInputView` - a method that measures its text-wise height and cleans up any show/hides it has to do in the process, then returns the measurements 🙂